src/LiveComponent/src/DependencyInjection/Compiler/OptionalDependencyPass.php

```diff
@@ -30,8 +31,14 @@ final class OptionalDependencyPass implements CompilerPassInterface
     public function process(ContainerBuilder $container): void
     {
         if ($container->hasDefinition('doctrine')) {
+            $doctrineArguments = [new IteratorArgument([new Reference('doctrine')])]; // TODO: add support for multiple entity managers
```

Comment:
I'd personally prefer keeping definitions separated, it's simpler to read/maintain

Reply:
I have reverted the change and separated it foreach register